docs/README.md

```diff
@@ -0,0 +1,79 @@
+# Core
+
+The core module provides the entry point for platform code. It includes functionality for retrieving information about the current environment, the module registry, plus parsing and reading the project configuration.
+
+**Note** all functions documented below are under the `HM\Platform` namespace.
+
+## Environment
+
+The following functions are available to use at any point on or after the `hm-platform.autoloader_loaded` action.
+
+**`get_environment_name() : string`**
```

Comment:
Yeah I did wonder. If we did have a TOC it could end up looking a bit odd with these. I've never found an approach to documenting functions in markdown that I really like tbh!

Reply:
Elsewhere I have functions under a 3rd level heading already so I think I'll keep it like this for now.